### PR TITLE
Closes #895: Create a new block in Imagify

### LINKED ISSUE #895: Create a New Block in Imagify
Description and assets available [here](https://docs.google.com/document/d/1LKH9s66lEYYveBus5imefMyuWGx1mnPUAHHogj9GO80/edit#heading=h.wmwupfxny905).

### PR DESCRIPTION
**Pull request by jeawhanlee**
# Description

Fixes #895 
*Explain how this code impacts users.*

## Type of change

- [x] New feature (non-breaking change which adds functionality).
- [x] Enhancement (non-breaking change which improves an existing functionality).

## Detailed scenario

Adds a new block for plugin family in settings page.

### Documentation

New package created, which is used to get the plugin family data for one click installation & activation.

### New dependencies

https://packagist.org/packages/wp-media/plugin-family

### Risks

N/A

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.

## Code style
- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

# Additional Checks
- [x] In the case of complex code, I wrote comments to explain it.
- [x] I added error handling logic when using functions that could throw errors (HTTP/API request, filesystem, etc.)

### CODE REVIEW

**Comment by jeawhanlee** [comment]
We have section showing plugins with different state ( Install, Activated & already Activated). Also, the activated plugins comes last in the order.

Screenshot 2024-10-21 at 12 46 26

File: inc/classes/class-imagify-views.php [hunk 1]
```diff
@@ -214,7 +215,14 @@ public function add_network_menus() {
 	 * @since 1.7
 	 */
 	public function display_settings_page() {
-		$this->print_template( 'page-settings' );
+		$plugin_family = new PluginFamily();
+		$plugin_family = $plugin_family->get_filtered_plugins( 'imagify-plugin/imagify' );
```
Comment (remyperona) [inline inc/classes/class-imagify-views.php]: It's a bit confusing when reading to re-assign the variable from a class instance to the value of the method return.

Maybe rename the second one to `$plugins_array` or something like that?

File: views/page-settings.php [hunk 2]
```diff
@@ -104,27 +104,40 @@
...
+								_e( 'Beyond Imagify, there\'s a whole family of plugins designed to help you build better, faster and safer websites. Each one is crafted with our unique blend of expertise, simplicity and
+outstanding support. Combine our plugins below to build incredible WordPress webistes!', 'imagify');
+								?>	
+							</p>
+							<?php foreach ( $data['plugin_family'] as $plugin => $plugin_data ) : ?>
+								<div class="imagify-plugin-family-col">
+									<div class="imagify-card">
+										<div class="imagify-card-header">
+											<div class="imagify-card-logo">
+												<img src="<?php echo IMAGIFY_ASSETS_IMG_URL . esc_attr( $plugin_data['logo']['file'] ); ?>" style="width: <?php echo esc_attr( $plugin_data['logo']['width'] ); ?>">
```
Comment (remyperona) [inline views/page-settings.php]:
```suggestion
												<img src="<?php echo esc_url( IMAGIFY_ASSETS_IMG_URL .  $plugin_data['logo']['file'] ); ?>" style="width: <?php echo esc_attr( $plugin_data['logo']['width'] ); ?>">
```

@piotrbak @DahmaniAdame Should we add optimization hints on those images? (like lazyload/decode/fetchpriority)

**Review by remyperona — CHANGES_REQUESTED** [review]

File: views/page-settings.php (hunk 2 above)
Comment (piotrbak) [reply views/page-settings.php]: `loading="lazy"` sounds very safe. For others let's wait and do it as a separate project for all the images

**Review by remyperona — APPROVED** [review]

**Comment by hanna-meda** [comment]
@jeawhanlee, thank you for your work on this. I've noted a few minor fixes below:
1. Commas needed in 2 places from description & "website" at the end is misspelled.
Screenshot 2024-10-28 at 16 07 24
2. Learn More links should open in new tab. Currently, they open in the same tab.

**Comment by jeawhanlee** [comment]
Thanks @hanna-meda 
Wordings has been fixed and `Learn more` links now open in a new page.

Screenshot 2024-10-28 at 16 36 05

**Comment by piotrbak** [comment]
@jeawhanlee Seems like we're good to go here

**Pull request merged**